### ISSUE TITLE
Getting aggregated data for displaying charts

### ISSUE THREAD

**Issue by xkuyax** [swharden/FftSharp#27]
Hi,

we would like to use this library to render your charts on the web. Is there any easy way to downsample/aggregate the data like you do in your charts? Something like .AggregateChart() which returns x: and y: which can be passed to different chart libraries or other tools

Thanks for your help!

**Comment by swharden** [comment]
Hi @xkuyax, 

I don't think I fully understand your question. This library is designed to accept data (e.g., `double[] xs` and `double[] ys`) and produce a `Bitmap` you can display (in a GUI) or save to disk (perhaps as a jpeg). I have used this library in Azure Functions to generate images to display on the web ([here](https://swhardendev.z13.web.core.windows.net/)).

If you would rather use another charting library to display `xs` and `ys` on the web, you can pass that data into the other library and do not need to use ScottPlot.

If your question relates to client-side interactive charting with JavaScript, I recommend looking into Google Charts https://developers.google.com/chart/interactive/docs/gallery

Let me know if this helps, or if I am misunderstanding your question! Thanks
Scott

**Comment by xkuyax** [comment]
Hi @swharden , thanks for your very fast reply!
I have been looking at your example program, which generates the fftPower, freqs and times arrays, and feeds this into the ScottPlot.Plot() 
I generated this myself for an wav File i got (30s, 44k SampleRate) which resulted in 1,4million doubles. Id rather not transmit that over the web, but instead calculate xs and ys (i guess thats the "downsampled" data?) and send this to my chart lib (and make an new call with different calculation parameters on scrolling to update the chart)

Basically what i want are 2 arrays:
X: (2.5 Hz, 5 Hz, 7.5 Hz,...)
Y: (-1, 2, 0, 1,) the fft values for the specified frequency in x (but averaged, like it is being down with your plot library)

Thanks!

**Comment by swharden** [comment]
Hey @xkuyax, 

First off let me apologize - I got mixed up and thought this question was posted as a ScottPlot issue and now I see it is a FftSharp issue. This is why I was confused, but your queston makes more sense to me now that I realize my mistake. If you analyze 30s of FFT data it will be very slow and produce a very large amount of data (you noted 1.4 million points). These types of high resolution FFTs are rarely required, and as you have found are difficult to display.

You indeed can indeed downsample the data (taking every Nth point and dividing sample rate by N) prior to calling the FFT methods to achieve fast calculations and a result with a small number of points.

Alternatively, you can also divide your signal into many 1s chunks, calculate all the individual FFTs, and average them together. This process is typically accompanied by a windowing function so each chunk starts and ends at 0.

You may find my Spectrogram library helpful, as its purpose is to simplify the process of creating many small FFTs from a long recording. https://github.com/swharden/Spectrogram

Let me know if this helps solve your problem!

**Comment by swharden** [comment]
Also, it may help to review the quickstart source code which produces this output

![image](https://user-images.githubusercontent.com/4165489/99289949-a8ff7f80-280b-11eb-83aa-6bda6da043c6.png)

https://github.com/swharden/FftSharp/blob/1e623b5aab99dbb60231a933c740b3df138caee2/src/FftSharp.Quickstart/Program.cs#L1-L68

You can see that `double[] freqs` and `double[] fftPower` and are the `Xs` and `Ys` you are probably trying to plot. Since they are simple arrays, you can plot them with any charting library. In the example I use ScottPlot, but alternatively you could send them to the browser to display interactively or something like that.

Hope it helps!
Scott
